app/views/api/proxy_rules/index.html.slim

```diff
@@ -0,0 +1,46 @@
+- content_for :sublayout_title, 'Mapping Rules'
+
+table#proxy-rules
+  thead
+    tr
+      th Verb
+      th Pattern
+      - if @service.using_proxy_pro?
+        th Redirect
+      th title="increment" +
+      th
+        ' Metric or Method
+        | (
+        = link_to 'Define', admin_service_metrics_path(@service), title: 'Define Methods & Metrics for this service'
+        | )
+      th Last?
+      th Position
+      th.actions
+        = link_to 'Add Mapping Rule', new_admin_service_proxy_rule_path(@service), class: 'action add'
+  tbody
+    - @proxy_rules.each do |rule|
+      tr
+        td.http_method
+          = rule.http_method
+        td.pattern
+          = rule.pattern
+          - if rule.pattern == '/'
+            span.fa.fa-exclamation-triangle.disabled#catch-all-warning title=(t('api.integrations.proxy.proxy_rule_catch_all_warning'))
+        - if @service.using_proxy_pro?
+          td.redirect_url
+            = rule.redirect_url
+        td.delta
+          = rule.delta
+        td.metric
+          = rule.metric.name
```

Comment:
I suggest this instead:
```suggestion
          = rule.metric_system_name
```

For 3 reasons:

1. the Law of Demeter. Let's just make calls to those classes it directly knows of 😄 Better not to make a chain of call from different places and then if we have to change it, we have to change it everywhere the chain is written.

2. `rule.metric.name` if metric is nil for any reason (although it shouldn't, but weirder things have happened 😄 ), it will raise a `NotMethodError`, but with `metric_system_name` we are already covering that case and it will return just nil:
https://github.com/3scale/porta/blob/8aed1303c2193d09b788696c4159c8cb9032770b/app/models/proxy_rule.rb#L92-L94

3. `system_name` and `name` are the same thing for metrics and there is a message in the `metric` model that I find very confusing but sounds to me like better not to use `name`:

https://github.com/3scale/porta/blob/8aed1303c2193d09b788696c4159c8cb9032770b/app/models/metric.rb#L78-L81

https://github.com/3scale/porta/blob/8aed1303c2193d09b788696c4159c8cb9032770b/app/models/metric.rb#L1

Reply:
makes total sense! thanks for it